### PR TITLE
Support multiple full recipes roots

### PR DESCRIPTION
**Pull request by jgleissner**
Instead of additional data roots only, support additional full recipes roots.
This allows to amend image definitions and add more schema templates.
Also changes obs_service to support arbitrary number of repos.

### CODE REVIEW

**Comment by jgleissner** [comment]
@aosthof As a heads-up, this will change the keg call from `keg -r keg-recipes -a keg-recipes-private/data ...` to `keg -r keg-recipes -r keg-recipes-private ...`.

File: kiwi_keg/image_definition.py [hunk 1]
```diff
@@ -64,19 +63,22 @@ def __init__(
...
-        if data_roots:
-            self._data_roots += data_roots
-        if not os.path.isdir(recipes_root):
-            raise KegDataError(
-                'Image Definition: {root} does not exist'.format(
-                    root=recipes_root
+        for recipes_root in self._recipes_roots:
+            if not os.path.isdir(recipes_root):
+                raise KegDataError(
+                    'Image Definition: {root} does not exist'.format(
```
Comment (jesusbv) [inline kiwi_keg/image_definition.py:69]: shouldn't it be `Recipes roots: {....`

File: kiwi_keg/obs_service/compose_kiwi_description.py [hunk 2]
```diff
@@ -68,18 +62,22 @@ def main() -> None:
...
+    temp_git_dirs = []
+    for repo, branch in itertools.zip_longest(args['--git-recipes'], args['--git-branch']):
+        temp_git_dir = Temporary(prefix='keg_recipes.').new_dir()
+        if branch:
+            Command.run(['git', 'clone', '-b', branch, repo, temp_git_dir.name])
+        else:
+            Command.run(['git', 'clone', repo, temp_git_dir.name])
+        temp_git_dirs.append(temp_git_dir.name)
 
+    print(temp_git_dirs)
```
Comment (jesusbv) [inline kiwi_keg/obs_service/compose_kiwi_description.py:77]: s/print/

File: kiwi_keg/image_definition.py [hunk 3]
```diff
@@ -64,19 +63,22 @@ def __init__(
         self._data = self._dict_type({})
         self._config_script = None
         self._images_script = None
-        if data_roots:
-            self._data_roots += data_roots
-        if not os.path.isdir(recipes_root):
-            raise KegDataError(
-                'Image Definition: {root} does not exist'.format(
-                    root=recipes_root
+        for recipes_root in self._recipes_roots:
```
[the remark this answers is not in the document]
Comment (jgleissner) [reply kiwi_keg/image_definition.py:66]: Not really sure what a decorator would be good for in this case tbh. Do you suggest to move the checks to a separate function? The __init__ function isn't that big. Not sure whether moving the checks to a separate function would be a big gain in terms of readability, but I don't feel strongly about it.

**Review by rjschwei — APPROVED** [review]

File: kiwi_keg/image_definition.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jgleissner) [reply kiwi_keg/image_definition.py]: Fair enough. I moved the checks into separate functions for better readability.

**Review by jesusbv — APPROVED** [review]
Thanks for the changes

LGTM !

**Pull request merged**